src/adhocracy_core/adhocracy_core/authentication/__init__.py

```diff
@@ -90,6 +93,46 @@ def wrapped_view(context, request):
     return wrapped_view
 
 
+def is_header_password(request: IRequest) -> bool:
+    """Check if request provided the password in the Password header."""
+    return UserPasswordHeader in request.headers
+
+
+def get_header_password(request: IRequest) -> str:
+    """Return the password in the Password header."""
+    if is_header_password(request):
+        return request.headers[UserPasswordHeader]
+
+
+def validate_password_header(view: callable):
+    """Decorator vor :term:`view` to check if the password header may be set.
+
+    :raise pyramid.httpexceptions.HTTPBadRequest: if password is invalid.
+    """
+    def wrapped_view(context, request):
+        password_is_set = is_header_password(request)
+        error = None
+        if password_is_set:
+            content = request.registry.content
+            user = request.user
+            password = get_header_password(request)
+            is_valid = user.is_password_valid(request.registry, password)
+            if not is_valid:
+                error = error_entry('header',
+                                    UserPasswordHeader,
+                                    'Invalid password')
+            is_required = content.is_password_required(context, request)
+            if not is_required:
+                error = error_entry('header',
+                                    UserPasswordHeader,
+                                    'Password not required')
+        if error:
+            request.errors.append(error)
+            raise HTTPBadRequest()
+        return view(context, request)
+    return wrapped_view
```

Comment:
`is_required` was not a good name for the variable/function.

At this point we can only tell that some sheets require a password. We do not know if this request wants to edit these sheets and cannot catch that case here.

At a later point during the processing of the sheets, the sheet permissions are checked and also that the password header is required but not set.